src/mongocxx/exception/error_category.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <string>
+
+#include <mongocxx/exception/error_category.hpp>
+
+#include <mongocxx/exception/mongoc_error.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+namespace exception {
+
+///
+/// A std::error_category for codes returned by libmongoc (via bson_error_t's).
+///
+class mongoc_error_category_impl final : public std::error_category {
```

Comment:
It makes sense to me to put it in an unnamed namespace, but just so I know what are the concrete benefits?